scss-style-guide.md

```diff
@@ -0,0 +1,100 @@
+# SCSS Style-Guide
+
+We use BEM with unlimited component nesting.
+
+## Glossar
+
+### Block `.button`
+
+### Element`.button__text`
+
+### Modifier`.button--green`
+
+
+
+## Ordering of styles in scss files
+
+The ordering of the elements should be related to the top to bottom occurances of the class in html.
+
+1. `@extend`
+2. `@include`
+3. Elements
+4. Modifiers
+
+### Example
+
+```
+.block {
+	// first extend
+	@extend .box;
+	// second include
+	@include arrow-mixin;
+	// third elements
+	&__child {
+	
+	}
+	// forth modifiers
+	&--blue {
+	
+	}
+}
+```
+
+
+
+# Rule one
+
+For each HTML-Element there should be at maximum one block-element.
+
+## Don't
+
+```html
+<div class="box grid__space__y"></div>
+```
+
+## Do
+
+```html
+<div class="my_custom_class"></div>
+```
+
+```scss
+.my_custom_class {
+	@extend .box, .grid__space__y;
+}
+```
```

Comment:
I understand your concern, but I am not certain if I agree with your solution. Since you incidentally picked a very abstract "don't" example, lets look at an example like `reservation-detail__content__section__data-list__description`. This essentially serves as an ID, and is very far from a component-based architecture.

I feel like we went from one extreme to the other. I, personally, don't mind class composition. It's transparent and can speed up development. What we decided against, was you next rule, cascading dependencies. Which I still believe to be a good idea because again, it promotes transparency.